### DISCUSSION TITLE
Turn morphology labels and wordbase into word

### DISCUSSION THREAD

**Discussion by schieslu** [explosion/spaCy#11883] [Help: Other Questions] [answered]
Hello everyone, I was wondering, if it is somehow possible to use the spacy morphologizer in a reverse way, so that I can input a label and a wordbase and get the word in the wanted morphology back. I am using the German Model de_core_news_large to extract noun chunks from a text. I need to "normalize" them, so that the extracted adjectives and nouns alle have the same grammatical Case. Any help is appreciated.

**Comment by polm** [comment, marked as answer]
No, you can't do that with the Morphologizer or Lemmatizer. Turning a word into a specific form is known as "realization" or "surface realization" and is an NLG feature. See [here](https://stackoverflow.com/questions/45590278/how-to-inverse-lemmatization-process-given-a-lemma-and-a-token) for some more details. 

If you need to normalize words to the same case, usually you'd use the dictionary form provided by the lemmatizer. Is there a reason that doesn't do what you want?

**Reply by schieslu** [reply]
Hi polm, thanks for your reply. To be honest, it didnt come to my mind to use the dictionary form of the lemmatizer. Could you elaborate on how that would work? Thank you!

**Reply by polm** [reply]
The lemma - available in `token.lemma_` - is the dictionary form.

**Reply by schieslu** [reply]
Thanks again for your reply. I tested the dictionary form and I think I can answer your question from before

> Is there a reason that doesn't do what you want?

The lemma alone is not what I need. I need a specific grammatikal case so that I can give the lemma of a word and the desired case as input and get the word in the desired case as output. That is the transformation I'm looking for. So I guess what I need is a dictionary containing all words in all of their cases and a function to pick me the desired one. Maybe this is not possible using Spacy though

**Reply by polm** [reply]
OK, in that case it sounds like you need a surface realizer, which may or may not use a dictionary. spaCy doesn't have support for that.
